Automattic/wp-calypso
# Stats: Introduce Tier Upgrade Notice for showing upgrade information

Related to #84671

## Proposed Changes

* Introduce the `TierUpgradeNotice` to show information for the upgrade flow.

## Testing Instructions



* Spin this change on the local Jetpack environment: `STATS_PACKAGE_PATH=/path-to-jetpack/projects/packages/stats-admin yarn dev`.
* Edit the code to disable `isCommercial` check for a non-commercial testing site. https://github.com/Automattic/wp-calypso/pull/84880/files#diff-bbc3bb485e5fb5d297cfdc0c767f67a313b15896520f130e68ad3292b40cecdbR134
* Navigate to the Stats > `Traffic` page with the feature flag for a site with Stats Commercial license: `/wp-admin/admin.php?page=stats#!/stats/day/{site-id}?page=stats&flags=stats%2Ftier-upgrade-slider`.
* Edit the code with different `currentUsage` to test for different notices. https://github.com/Automattic/wp-calypso/pull/84880/files#diff-f018bb72763ca969cc57532cf3aa3f75c77cff12eb7589d101dd34de956b4ae0R21
* Ensure the Tier Upgrade Notice shows messages according to the current usage.

<img width="1257" alt="截圖 2023-12-06 下午11 35 14" src="https://github.com/Automattic/wp-calypso/assets/6869813/2047a973-50eb-46d3-b241-cd23da2d6c7e">

<img width="1257" alt="截圖 2023-12-06 下午11 35 52" src="https://github.com/Automattic/wp-calypso/assets/6869813/4a4237d0-b4a6-4b32-a00c-74786d5bd98a">

## Pre-merge Checklist
